zc_install/includes/classes/class.systemChecker.php

```diff
@@ -344,7 +344,14 @@ public function checkPhpVersion($parameters)
   public function checkHtaccessSupport($parameters)
   {
     $testPath = $_SERVER['HTTP_HOST'] . $_SERVER['SCRIPT_NAME'];
-    $testPath = 'http://' . substr($testPath, 0, strpos($testPath, '/zc_install')) . '/includes/filenames.php';
+    $proto = 'http://'; 
+
+    if (isset($_SERVER['REQUEST_SCHEME']) && $_SERVER['REQUEST_SCHEME'] == 'https') { 
+      $proto = "https://"; 
+    } else if (isset($_SERVER['SCRIPT_URI']) && strncmp($_SERVER['SCRIPT_URI'], 'https', 5) == 0) { 
```

Comment:
Let's take a different approach:
- since `$request_type` is already set to `SSL` when all the usual detections pass, let's use that instead of all this new code
- use that to determine whether we test https first
- then if that first test fails, fallback and re-test with the other

Reason: If the server can't be trusted to do its own 301 redirect to a proper 403 response (and not a 404 because the 403 isn't fully configured), we probably can't trust it a binary either/or situation either, so a fallback is the only option. 

Or maybe we just remove this check altogether and stop taking responsibility for exposing the host's misconfiguration.

Reply:
YAY!  I like door number three.

Reply:
What impact will this "removal" have on PCI of the product (ZC) overall? Its one thing for a server to be misconfigured and prone to its own issues and another for performing the check and identifying the strengths/weaknesses of performing the check with additional information about how to resolve...

Reply:
> > Or maybe we just remove this check altogether and stop taking responsibility for exposing the host's misconfiguration.

> What impact will this "removal" have on PCI of the product (ZC) overall? Its one thing for a server to be misconfigured and prone to its own issues and another for performing the check and identifying the strengths/weaknesses of performing the check with additional information about how to resolve...

None. Zen Cart isn't responsible for the server's actual configuration. That's up to the hosting company. Just like a properly functioning SSL certificate chain isn't something Zen Cart can fix or adjust or otherwise do anything about.

Reply:
Ditto.  Zen Cart also doesn't check for modified RPMs.  Out of scope.

Reply:
Well, I had come up with this based on the earlier discussion/advice of @drbyte.

````php
  public function checkHtaccessSupport($parameters)
  {
    global $request_type;
    $testPath = $_SERVER['HTTP_HOST'] . $_SERVER['SCRIPT_NAME'];
    $testPath2 = substr($testPath, 0, strpos($testPath, '/zc_install')) . '/includes/filenames.php';
    $testPath_array = [1 => 'http://', 2 => 'https://'];
    $testPath = ($request_type != 'SSL' ? $testPath_array[1] : $testPath_array[2]) . $testPath2;
    
    if (!function_exists('curl_init')) {
      return TRUE;
    }

    $result = TRUE;
    $repeat = 2;
    while ($repeat) {
      $repeat--;
      $resultCurl = self::curlGetUrl($testPath);
      if (!(isset($resultCurl['http_code']) && $resultCurl['http_code'] == '403'))
      { // test again with redirects enabled
        $resultCurl = self::curlGetUrl($testPath, true);
        if (!(isset($resultCurl['http_code']) && $resultCurl['http_code'] == '403'))
        {
          if ($request_type == 'SSL' && $repeat) {
            $testPath = $testPath_array[$repeat] . $testPath2;
            continue;
          }
          $result = FALSE;
        }
      }
      break;
    }

    return $result;
  }
````

Reply:
Hmmm ... ya, maybe something like that is fine.
I'm inclined to drop the 2 test URLs into an array, with https first, then http, then early-return as soon as one passes. (still doing the redirect fallback).  Then can drop the $repeat var and the $result boolean

Reply:
This would be a bunch more testing.  Do you have the time and resources (webserver that can be switched in and out of forced https) to do that?

Reply:
The test would need to be done by those on the forum complaining that it's "broken" currently. Granted, I still think the real issue is a cPanel problem. I can't test cPanel though.

Reply:
It's never been a "hard fail". This check only sets up a warning/alert for consideration.

If an update prevents "false positives", great. But if the server has a misconfiguration we can't detect that without making this code overly complicated. And we can never "fix" it because it's out of scope.